src/drivers/i2c/i2c.h

```diff
@@ -0,0 +1,22 @@
+#ifndef I2C_H
+#define I2C_H
+
+#include "rocketlib/include/common.h"
+#include <stdint.h>
+
+typedef enum {
+    I2C_DEVICE_POLULU, // Polulu AltIMU-v6
+    I2C_DEVICE_MOVELLA, // Movella mti-630 IMU
+    I2C_DEVICE_ST, // ST LSM6DSO32 IMU
+    I2C_DEVICE_COUNT // Number of devices
+} i2c_device_t;
+
+// Must be called before RTOS scheduler starts
+w_status_t i2c_init(i2c_device_t device, uint32_t timeout_ms);
```

Comment:
comment more. dont expect users to reference the design doc when using this function, it should be explained fully in code. (what it does, what each param is etc)